Zabolotskiy/hash/main.cpp

```diff
@@ -0,0 +1,64 @@
+#include <Windows.h>
+#include <fstream>
+#include "../src/hashlibpp.h"
+using namespace std;
+
+HANDLE inputMutex, outputMutex;
+ifstream *infile;
+ofstream *outfile;
+hashwrapper *md5 = new md5wrapper();
+hashwrapper *sha1 = new sha1wrapper();
+char* block;
+int sizeBlock = 0;
+char* hashType;
+
+void Thread(void* pParams)
+{
+		block = new char[sizeBlock];
+		while (!infile->eof())
+		{
+			WaitForSingleObject(inputMutex, INFINITE);
+				infile->read((char*)(block), sizeBlock);
+			ReleaseMutex(inputMutex);
+			
+			WaitForSingleObject(outputMutex, INFINITE);
+				if (!strcmp(hashType, "md5"))	*outfile << md5->getHashFromString(block) << endl;
+				else	*outfile << sha1->getHashFromString(block) << endl;
+			ReleaseMutex(outputMutex);
+		}
+}
+
+int main ( int argc, char **argv)
+{
+	char outName[300];
+	int countThreads = 1;
+	if (!strcmp(argv[4], "sync")) countThreads = 3;
```

Comment:
In general, same comments as for Kristy's lab :).
Here and a little below - crashes (here because argv may be empty, below because the files might not exist)